code/modules/food_and_drinks/food/pizza.dm

```diff
@@ -0,0 +1,361 @@
+	slice_type = null
+
+/obj/item/food/pizza/raw/MakeBakeable()
+	AddComponent(/datum/component/bakeable, /obj/item/food/pizza, rand(70 SECONDS, 80 SECONDS), TRUE, TRUE)
+
+/obj/item/food/pizzaslice
+	icon = 'icons/obj/food/pizzaspaghetti.dmi'
+	food_reagents = list(/datum/reagent/consumable/nutriment = 5)
+	foodtypes = GRAIN | DAIRY | VEGETABLES
+	w_class = WEIGHT_CLASS_SMALL
+
+
+/obj/item/food/pizzaslice/Initialize()
+	. = ..()
+	//AddElement(/datum/element/processable, TOOL_ROLLINGPIN, /obj/item/stack/sheet/pizza, 1, 10)
+
+/obj/item/food/pizza/margherita
+	name = "pizza margherita"
+	desc = "The most cheezy pizza in galaxy."
+	icon_state = "pizzamargherita"
+	food_reagents = list(/datum/reagent/consumable/nutriment = 25, /datum/reagent/consumable/tomatojuice = 6, /datum/reagent/consumable/nutriment/vitamin = 5)
+	tastes = list("crust" = 1, "tomato" = 1, "cheese" = 1)
+	tastes = list("crust" = 1, "tomato" = 1, "cheese" = 1)
+	foodtypes = GRAIN | VEGETABLES | DAIRY
+	slice_type = /obj/item/food/pizzaslice/margherita
+
+/obj/item/food/pizza/margherita/Initialize()
+	. = ..()
+	AddElement(/datum/element/processable, TOOL_KNIFE, /obj/item/food/pizzaslice/margherita, 6, 30)
+
+/obj/item/food/pizza/margherita/raw
+	name = "raw pizza margherita"
+	icon_state = "pizzamargherita_raw"
+	foodtypes =  GRAIN | VEGETABLES | DAIRY | RAW
+	burns_in_oven = FALSE
+	slice_type = null
+
+/obj/item/food/pizza/margherita/raw/MakeBakeable()
+	AddComponent(/datum/component/bakeable, /obj/item/food/pizza/margherita, rand(70 SECONDS, 80 SECONDS), TRUE, TRUE)
+
+/obj/item/food/pizza/margherita/robo
+	food_reagents = list(/datum/reagent/nanomachines = 70, /datum/reagent/consumable/nutriment = 25, /datum/reagent/consumable/tomatojuice = 6, /datum/reagent/consumable/nutriment/vitamin = 5)
+
+/obj/item/food/pizzaslice/margherita
+	name = "margherita slice"
+	desc = "A slice of the most cheezy pizza in galaxy."
+	icon_state = "pizzamargheritaslice"
+	tastes = list("crust" = 1, "tomato" = 1, "cheese" = 1)
+	foodtypes = GRAIN | VEGETABLES | DAIRY
+
+/obj/item/food/pizzaslice/margherita/Initialize()
+	. = ..()
+	AddComponent(/datum/component/customizable_reagent_holder, null, CUSTOM_INGREDIENT_ICON_FILL, max_ingredients = 12)
+
+/obj/item/food/pizza/meat
+	name = "meatpizza"
+	desc = "Greasy pizza with delicious meat."
+	icon_state = "meatpizza"
+	food_reagents = list(/datum/reagent/consumable/nutriment = 25, /datum/reagent/consumable/tomatojuice = 6, /datum/reagent/consumable/nutriment/vitamin = 8)
+	foodtypes = GRAIN | VEGETABLES| DAIRY | MEAT
+	slice_type = /obj/item/food/pizzaslice/meat
+
+/obj/item/food/pizza/meat/raw
+	name = "raw meatpizza"
+	icon_state = "meatpizza_raw"
+	foodtypes =  GRAIN | VEGETABLES| DAIRY | MEAT | RAW
+	burns_in_oven = FALSE
+	slice_type = null
+
+/obj/item/food/pizza/meat/raw/MakeBakeable()
+	AddComponent(/datum/component/bakeable, /obj/item/food/pizza/meat, rand(70 SECONDS, 80 SECONDS), TRUE, TRUE)
+
+/obj/item/food/pizza/meat/Initialize()
```

Comment:
Redundant due to the base typepath having a generic method, would actually create double.